app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ActiveRecord::Base
+    has_secure_password
+    before_save {self.email = email.downcase}
+
+    validates :username, presence: true,uniqueness: true,length: {minimum: 5, maximum: 25}
+    VALID_EMAIL_REGEX= /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+    validates :email, presence: true, length:{maximum: 105},uniqueness: true,format: {with: VALID_EMAIL_REGEX}
```

Comment:
You do not need valid_email_regex if you include `type='email'` for the input in the form. Same thing with the format options for validates.